### ISSUE TITLE
no-member false positive for custom enum class

### ISSUE THREAD

**Issue by mitar** [PyCQA/pylint#2422]
### Steps to reproduce

Have two files.

`utils.py`:

```python
import enum

class Enum(enum.Enum):
    pass
```

`test.py`:

```python
import utils

class Foo(utils.Enum):
    A = 1

print(Foo.A.name)
```

### Current behavior

Running `pylint test.py` outputs:

```
test.py:8:6: E1101: Instance of 'int' has no 'name' member (no-member)
```

### Expected behavior

This error should not be raised.

Interestingly, if `Enum` is defined in the same file, this does not happen.

### pylint --version output

```
pylint 2.1.1
astroid 2.0.3
Python 3.6.3 (default, Oct  3 2017, 21:45:48) 
[GCC 7.2.0]
```

**Comment by Kilo59** [comment]
I think I was having a similar issue.
I temporarily fixed mine by renaming my `utils` module to something other than utils.
Seems to be some kind of name collision when using the `utils` name. 

I'd like to find a better solution.

**Comment by PCManticore** [comment]
@mitar This is most likely a duplicate of https://github.com/PyCQA/pylint/issues/533. The gist of this issue is that right now pylint cannot understand subclasses of Enums in the same way it understands an enum, resulting in this false positive you're seeing.

**Comment by mitar** [comment]
But if it is in the same file, then it works?

**Comment by PCManticore** [comment]
I'm still getting the same error even in the same file, using the latest pylint.

**Comment by mitar** [comment]
I have `pylint 2.1.1` and if I run `pylint test.py`, with `test.py`:

```
import enum

class Enum(enum.Enum):
    pass

class Foo(Enum):
    A = 1

print(Foo.A.name)
```

The only output is:

```
************* Module test
test.py:1:0: C0111: Missing module docstring (missing-docstring)
test.py:3:0: C0111: Missing class docstring (missing-docstring)
test.py:6:0: C0111: Missing class docstring (missing-docstring)

---------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: -13.33/10, +18.33)
```

**Comment by PCManticore** [comment]
Gotcha, I've been testing with a different name altogether. 

This works in the same file because it falls under this check https://github.com/PyCQA/astroid/blob/b8122df810a7d3e1664154fdcb9873dcfb6e2ecb/astroid/brain/brain_namedtuple_enum.py#L269, since it has the same name as one we're expecting to have for an `Enum` class. This explains why it works in the same file but it doesn't when imported from a different file.
Unfortunately as the comment from the linked file mentions, that code is just a hack to support enums and it's very brittle. Ideally, we'd be able to infer that the class is a subclass of an `Enum`, without looking explicitly for the basenames, but the reason we're not doing that now is that it's going to be done for each class that astroid is going to find, so could be a potential performance bottleneck. This issue should be addressed by a solution for the original issue.
